pintracker/optracker/operationtracker.go

```diff
@@ -159,3 +221,64 @@ func (opt *OperationTracker) GetOpContext(c *cid.Cid) context.Context {
 	}
 	return op.Context()
 }
+
+// Filter returns a slice of api.PinInfos that had associated
+// Operations that matched the provided filter. Note, only supports
+// filters of type OperationType or Phase, any other type
+// will result in a nil slice being returned.
+func (opt *OperationTracker) Filter(filters ...interface{}) []api.PinInfo {
+	var pinfos []api.PinInfo
+	opt.mu.RLock()
+	defer opt.mu.RUnlock()
+	ops := filterOpsMap(opt.operations, filters)
+	for _, op := range ops {
+		pinfos = append(pinfos, opt.unsafePinInfo(op))
+	}
+	return pinfos
+}
+
+// FilterOps returns a slice that only contains operations
+// with the matching filter. Note, only supports
+// filters of type OperationType or Phase, any other type
+// will result in a nil slice being returned.
+func (opt *OperationTracker) FilterOps(filters ...interface{}) []*Operation {
```

Comment:
Per comments above/below, FilterOps should not be necessary.

Reply:
i.e. should be replaced by `CleanErrors` which atomically filters and cleans errors.

I will insist forever, there should be no way to get an operation out of the tracker (except when creating one).